app/models/user.rb

```diff
@@ -14,11 +14,13 @@ class User < ApplicationRecord
     validates :address, presence: true
     validates :city, presence: true, format: {with: /[A-Z]\D+/, message: "is invalid, please use Uppercase in first letter"}
     validates :postalcode, presence: true, format: {with: /\b\d{5}\b/, message: "is invalid, please enter a correct Postal Code"}
+
     validate  :is_adult?
 
     enum :doc_kind, {dni: 0, driver_license: 1, passport: 2}
+    enum :role, {admin: 00, customer: 01, premium_customer: 10, blocked: 0, banned: 1}
```

Comment:
almost perfect. except for a detail. what's the difference between 00 and 0 or 1 and 01?

I propose
banned 0
blocked 1
admin 10
customer 20
premium 30

Reply:
Honestly, thats a better way to name it. I wanted to give it a binary comprehensive/look. But that may lead to confusions